src/Process.php

```diff
@@ -120,6 +126,67 @@ public function start(LoopInterface $loop, $interval = 0.1)
     }
 
     /**
+     * This is a special implementation of the start method for the Windows operating system.
+     * This is needed because windows has a broken implementation of stdout / stderr pipes that
+     * cannot be fixed by PHP.
+     *
+     * @param LoopInterface $loop        Loop interface for stream construction
+     * @param float         $interval    Interval to periodically monitor process state (seconds)
+     * @throws RuntimeException If the process is already running or fails to start
+     */
+    protected function startWindows(LoopInterface $loop, $interval = 0.1)
+    {
+        $cmd = $this->cmd;
+
+        $stdoutName = tempnam(sys_get_temp_dir(), "out");
+        $stderrName = tempnam(sys_get_temp_dir(), "err");
+
+        // Let's open 2 file pointers for both stdout and stderr
+        // One for writing, one for reading.
+        $stdout = fopen($stdoutName, "w");
+        $stderr = fopen($stderrName, "w");
+        $stdoutRead = fopen($stdoutName, "r");
+        $stderrRead = fopen($stderrName, "r");
+
+
+        $fdSpec = array(
+            array('pipe', 'r'),
```

Comment:
Not sure if the analysis in #9 also applies to STDIN. Assuming it does, this means that we will still block when writing *more* data to the STDIN pipe as there's no way to detect when the receiving buffer is full?